include/zephyr/bluetooth/buf.h

```diff
@@ -169,15 +163,14 @@ void bt_buf_rx_freed_cb_set(bt_buf_rx_freed_cb_t cb);
  *  This will set the buffer type so bt_buf_set_type() does not need to
  *  be explicitly called.
  *
- *  @param type    Type of buffer. Only BT_BUF_CMD, BT_BUF_ACL_OUT or
- *                 BT_BUF_H4, when operating on H:4 mode, are allowed.
+ *  @param type    Type of buffer. BT_BUF_CMD or BT_BUF_ACL_OUT.
  *  @param timeout Non-negative waiting period to obtain a buffer or one of the
  *                 special values K_NO_WAIT and K_FOREVER.
  *  @param data    Initial data to append to buffer.
  *  @param size    Initial data size.
  *  @return A new buffer.
  */
-struct net_buf *bt_buf_get_tx(enum bt_buf_type type, k_timeout_t timeout,
+struct net_buf *bt_buf_get_tx(uint8_t type, k_timeout_t timeout,
```

Comment:
We need to maintain some level of type information here. Either we should rename the argument to `h4_type`, or seeing as the enum is packed, we keep using the enum here. I suggest adding `BUILD_ASSERT(sizeof(enum bt_buf_type) == sizeof(uint8_t))` in the latter case.

Reply:
> We need to maintain some level of type information here. Either we should rename the argument to `h4_type`, or seeing as the enum is packed, we keep using the enum here. I suggest adding `BUILD_ASSERT(sizeof(enum bt_buf_type) == sizeof(uint8_t))` in the latter case.

I think it'd be nice to try to use the enum everywhere, combined with the build assert.

Reply:
I moved from `uint8_t` to `enum bt_buf_type` for these APIs.